### DISCUSSION TITLE
how to generate sitemap.xml

### DISCUSSION THREAD

**Discussion by hristopanev** [quasarframework/quasar#12189] [Quasar App Extensions] [answered]
can i generate google sitemap.xml directly from the quasar?

**Comment by MilosPaunovic** [comment, marked as answer]
Short answer is no. Not using only Quasar. You have to use some external library or create it yourself.
I've used [sitemap-webpack-plugin](https://www.npmjs.com/package/sitemap-webpack-plugin) package for some of my projects and it worked like a charm.

**Reply by hristopanev** [reply]
Can i also generate dynamic links with it?

**Reply by MilosPaunovic** [reply]
When I was doing it, I had to create a custom XHR request to collect all the links from the page, so that's an idea for you to try in your code.

**Reply by hristopanev** [reply]
super thank you! I'll try :)

**Reply by hristopanev** [reply]
Hi Milos,
I failed with the implementation .. do you have an example for that.  :)
thank you :)

**Reply by MilosPaunovic** [reply]
Can't share actual code, but the idea behind it is that I had an endpoint which returned me the list of products needed, extracted URLs from there and added 'em to the `paths` array.

What exactly is troubling you, if you start with something and get to a specific problem, maybe then I'll be able to guide you better?

**Reply by hristopanev** [reply]
Hi Milos,
I give up with frontend sitemap. I make my sitemap.xml direct from backend. but now, how to register the link as a router? how to get example.com/sitemap.xml (that from backend direct, without vue component), before he redirect me to not found? Do you have Tip?

Thank you :)

**Reply by MilosPaunovic** [reply]
You need to upload your `sitemap.xml` file to server where you're deploying FE application.
